engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelDeepRenameTransform.kt

```diff
@@ -152,24 +155,34 @@ internal class NadelDeepRenameTransform : NadelTransform<NadelDeepRenameTransfor
         field: ExecutableNormalizedField,
         state: State,
     ): NadelTransformFieldResult {
+        val objectTypesNoRenames = field.objectTypeNames.filterNot { it in state.instructionsByObjectTypeNames }
+
         return NadelTransformFieldResult(
-            newField = null,
-            artificialFields = state.instructions.map { (coordinates, instruction) ->
+            newField = objectTypesNoRenames
```

Comment:
So a common "bug" we have is that we never consider cases where the field being transformed is in an interface, and that other fields in the interface implementations don't have that transform e.g.

```graphql
interface World {
  hello: String
}
type Earth implements World {
  hello: String
}
type Mars implements World {
  hello: String @renamed(from: "core.hello")
}
```

In this case the underlying query needs to be

```graphql
query {
  world {
    ... on Earth {
      hello
    }
    ... on Mars {
      core {
        hello
      }
    }
  }
}
```

But without this change we only send down


```graphql
query {
  world {
    ... on Mars {
      core {
        hello
      }
    }
  }
}
```